build.go

```diff
@@ -621,6 +630,55 @@ func buildSnap(target target) {
 	runPrint("snapcraft")
 }
 
+func shouldBuildSyso(dir string) string {
+	jsonPath := filepath.Join(dir, "versioninfo.json")
+	file, err := os.Create(filepath.Join(dir, "versioninfo.json"))
+	if err != nil {
+		log.Printf("Warning: unable to create %s: %v. Windows binaries will not have file information encoded.", jsonPath, err)
+		return ""
+	}
+
+	major, minor, patch, build := semanticVersion()
+	fmt.Fprintf(file, `{
+    "FixedFileInfo": {
+        "FileVersion": {
+            "Major": %s,
+            "Minor": %s,
+            "Patch": %s,
+            "Build": %s
+        }
+    },
+    "StringFileInfo": {
+        "FileDescription": "Open Source Continuous File Synchronization",
+        "LegalCopyright": "The Syncthing Authors",
+        "ProductVersion": "%s",
+        "ProductName": "Syncthing"
+    },
+    "IconPath": "assets/logo.ico"
+}`, major, minor, patch, build, getVersion())
+	file.Close()
+	defer func() {
+		if err := os.Remove(jsonPath); err != nil {
+			log.Printf("Warning: unable to remove generated %s: %v. Please remove it manually.", jsonPath, err)
+		}
+	}()
+
+	sysoPath := filepath.Join(dir, "cmd", "syncthing", "resource.syso")
+
+	runPrint("goversioninfo", "-o", sysoPath)
```

Comment:
Probably want to replace with runError

Reply:
Updated to use `runError` and also had `shouldBuildSyso return an error (could probably just rename to `buildSyso`) if it failed to build the syso.

**Testing**

1. Removed `goversioninfo` from my system and re-ran `go run build.go`.
   - Build succeeded with a logged warning
1. Added `goversioninfo` back to my system and re-ran `go run build.go`
   - Build succeeded without a warning and the `syncthing.exe` was encoded properly with the icon and build information.